### ISSUE TITLE
[Performance] NopResourceDisplayNameAttribute uses async methods synchronously - which leads to threadpool starvation in high traffic

### ISSUE THREAD

**Issue by kiranbhadani** [nopSolutions/nopCommerce#7092]
nopCommerce version: 
4.5, 4.6

Steps to reproduce the problem: 
We noticed issue in our production enviornment that site becomes unresponsive in some peek load, we have captured memory dump in the event scenario and during analysis found that many threads are waiting in NopResourceDisplayNameAttribute.DisplayName.get()

![image](https://github.com/nopSolutions/nopCommerce/assets/14998556/d1323a70-9638-4847-8c70-1680a064e692)
 
Problem is DisplayName property calls LocalizationService.GetResourceAsync() and waits for result

![image](https://github.com/nopSolutions/nopCommerce/assets/14998556/7e90918d-c537-4568-b882-2801fd7ff7f5)

I think possible solution could be to not load localized resource from the NopResourceDisplayNameAttribute.DisplayName instead handle it from NopLabelTagHelper.ProcessAsync so that it keeps all async stack

**Comment by skoshelev** [comment]
Closed by https://github.com/nopSolutions/nopCommerce/commit/b4f46d7be5248fd0a7efc2b0dc89dc1efea46172

**Comment by kiranbhadani** [comment]
@skoshelev I have just checked the commit and I see it's still using localization service call synchronously from  NopResourceDisplayNameAttribute, I am not sure if that being used or not but if it's still there, the threadpool issue will not get fixed

   //get locale resource value
            _resourceValue = EngineContext.Current.Resolve<ILocalizationService>().GetResourceAsync(ResourceKey).Result;

**Comment by skoshelev** [comment]
@kiranbhadani That's right, the call remains inside the attribute for now, in case the attribute is used somewhere else outside the tag helper (for example, in third-party plugins), but in our code we removed all calls to this property

**Comment by RomanovM** [comment]
@skoshelev Could we make the [same refactoring](https://github.com/nopSolutions/nopCommerce/commit/b4f46d7be5248fd0a7efc2b0dc89dc1efea46172#diff-13b59885f1b820461c41dfa6a72351928e015ff4b7f34b18ea6ebd1bc76281fb) for LabelTagHelper used in public store?

**Comment by skoshelev** [comment]
@RomanovM Yes, I think there will be no problem

**Comment by skoshelev** [comment]
Closed #7092
